core/src/main/java/com/linecorp/armeria/internal/common/ByteBufBinaryData.java

```diff
@@ -270,15 +246,15 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
-        if (!(obj instanceof HttpData)) {
+        if (!(obj instanceof ByteBufBinaryData)) {
             return false;
         }
 
         if (obj == this) {
             return true;
         }
 
-        final HttpData that = (HttpData) obj;
+        final ByteBufBinaryData that = (ByteBufBinaryData) obj;
```

Comment:
Should accept any `BinaryData` like `ByteBufHttpData` accepted any `HttpData`. (like how Java collections API works)